cache/network/download.go

```diff
@@ -23,31 +23,31 @@ var ErrCacheNotFound = errors.New("no cache archive found for the provided keys"
 
 // Download archive from the cache API based on the provided keys in params.
 // If there is no match for any of the keys, the error is ErrCacheNotFound.
-func Download(params DownloadParams, logger log.Logger) error {
+func Download(params DownloadParams, logger log.Logger) (matchedKey string, err error) {
```

Comment:
The diff may be making this difficult to read, but do we actually use the named return parameters?  It looks like each return statement populates them directly.  Should we remove the names?

Reply:
I wanted to use named parameters to make it more explicit what that returned `string` is. I realize this is not the primary reason for using named returns, so I'm happy to revert. Thoughts?

Reply:
Of all of the above options, the typealias seems the most appropriate option for me. I'll leave this untouched for now though, I'll need to edit this part of the code in a follow-up PR anyways, I'll  make the change then.